mail/dovecot/Portfile

```diff
@@ -1,110 +1,456 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
 PortSystem          1.0
+PortGroup           github 1.0
 
+github.setup        dovecot core 2.3.9.2
 name                dovecot
+revision            0
 conflicts           dovecot2
-epoch               20060722
```

Comment:
But we're changing Portfile names to eliminate the obsolete dovecot version 1 port, so the `epoch` no longer applies to the new `dovecot` (version 2) Portfile. There is no `epoch` in the current `dovecot2` Portfile; therefore, there should be no `epoch` in the renamed `dovecot` Portfile.

I don't believe `epoch` this applies in this scenario.

Reply:
Okay thanks. I presume the new epoch should be today's date?

Reply:
> Anyone who currently has dovecot installed will never be prompted to update to version 2.x if the epoch is removed.

I'll add the `epoch` statement, but wouldn't anyone who has dovecot version 1 installed see the latest update to version 2.3.9.2 and be automatically updated?

Also, no one should have dovecot version 1 installed, and there should be some faster process for obsoleting dangerously out-of-date ports.

Reply:
Also, what happens with `epoch` and the subport `dovecot-sieve`?

Reply:
Epoch is effectively the most significant part of the overall version. If you have epoch=0,version=1.0, then epoch=0,version=2.0 is newer. But epoch=1,version=1.0 is newer than epoch=0,version=2.0. The whole reason the epoch variable exists is so if a project changes its versioning scheme, you can specify that a version that looks older is actually newer.

There's no real reason to use a date for the epoch, but once it's added it can never be decreased (or removed, since the default value is 0). You don't need to increase the epoch in this instance because the new version looks bigger than the old one.

dovecot-sieve didn't exist previously, so any epoch will do. You can reset it to 0 in the subport or just inherit the epoch from the main port. Either will work fine.